cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -1015,6 +1016,11 @@ public void createOrDeleteReplicas(Map<Short, Set<String>> topicsByReplicationFa
             // Make sure the leader replica is in new replica list.
             newAssignedReplica.add(partitionInfo.leader().id());
             for (Node node : partitionInfo.replicas()) {
+              // Current clusterModel may have inconsistency with the partitionInfo. In that case, we skip the replica.
+              if (!brokerIds.contains(node.id())) {
+                continue;
+              }
+
```

Comment:
I think this change might be missing some inconsistencies in case brokers are part of the cluster, but the cluster metadata is inconsistent with cluster model.
Why don't we compare the equality of replicas of the target partition in (1) `ClusterModel` and (2) `Cluster`?

In case there is a mismatch, we can log this as a warn, explain that the fix is skipped due to stale cluster metadata, and skip the fix for that partition. With this approach, ss long as the topic_anomaly self-healing is enabled or if the user retries, any inconsistency in replication factor should eventually be fixed and we would not block the progress of fixes for other partitions.

^ We should also document this potential behavior in the JavaDoc of `UpdateTopicConfigurationRunnable` and `LiUpdateTopicConfigurationRunnable`

Reply:
That's actually a better idea. Let me make the change